source/quark/R/01_QuarkData.R

```diff
@@ -472,11 +481,71 @@ QuarkData$methods(
         data[ , targets] <<- tmp2
     },
 
-    cleanCollinVars = function(x)                                               {
+    
+   cleanCollinVars  = function(x)                                               {
         "Remove one variable from all collinear pairs"
-        collinVars <<- x
-        removeVars(x = unique(collinVars$var1), reason = "collinear")
-    },
+        collinVars     <<-  x
+        collinVarPairs <- collinVars[, 1:2]
+		
+        while(nrow(collinVarPairs) > 0){
+    
+        varCount     <- data.frame(table(unlist(collinVarPairs)))
+	maxVarCount  <- varCount[which(varCount$Freq == max(varCount$Freq)), ]
+	maxVarCommon <- intersect(names(respCounts), 
+		                          maxVarCount[, 1])
+    
+        ## Check for missing value counts if maximum counts are equal
+        if(nrow(maxVarCount) > 1){
+      
+        maxNaCountValue  <- max(respCounts[maxVarCommon])
```

Comment:
I am finding the variables in "maxVarCommon" that exist in both respCounts and maxVarCounts and then the maximum missing count value out of those and store in maxNaCountValue.